src/services/doppler-billing-user-api-client.ts

```diff
@@ -0,0 +1,140 @@
+import { ResultWithoutExpectedErrors, EmptyResultWithoutExpectedErrors } from '../doppler-types';
+import { AxiosInstance, AxiosStatic } from 'axios';
+import { AppSession } from './app-session';
+import { RefObject } from 'react';
+
+export interface DopplerBillingUserApiClient {
+  getBillingInformationData(): Promise<ResultWithoutExpectedErrors<BillingInformation>>;
+  updateBillingInformation(values: any): Promise<EmptyResultWithoutExpectedErrors>;
+}
+
+interface DopplerBillingUserApiConnectionData {
+  jwtToken: string;
+  email: string;
+}
+
+export interface BillingInformation {
+  sameAddressAsContact: boolean;
+  firstname: string;
+  lastname: string;
+  address: string;
+  city: string;
+  province: string;
+  country: string;
+  zipCode: string;
+  phone: string;
+  chooseQuestion: string;
+  answerQuestion: string;
+}
+
+export interface Features {
+  contactPolicies: boolean;
+}
+
+export class HttpDopplerBillingUserApiClient implements DopplerBillingUserApiClient {
+  private readonly axios: AxiosInstance;
+  private readonly baseUrl: string;
+  private readonly connectionDataRef: RefObject<AppSession>;
+
+  constructor({
+    axiosStatic,
+    baseUrl,
+    connectionDataRef,
+  }: {
+    axiosStatic: AxiosStatic;
+    baseUrl: string;
+    connectionDataRef: RefObject<AppSession>;
+  }) {
+    this.baseUrl = baseUrl;
+    this.axios = axiosStatic.create({
+      baseURL: this.baseUrl,
+    });
+    this.connectionDataRef = connectionDataRef;
+  }
+
+  private getDopplerBillingUserApiConnectionData(): DopplerBillingUserApiConnectionData {
+    const connectionData = this.connectionDataRef.current;
+    if (
+      !connectionData ||
+      connectionData.status !== 'authenticated' ||
+      !connectionData.jwtToken ||
+      !connectionData.userData
+    ) {
+      throw new Error('Doppler Billing User API connection data is not available');
+    }
+    return {
+      jwtToken: connectionData.jwtToken,
+      email: connectionData.userData.user.email,
+    };
+  }
+
+  private hasAddressInformation(billingInformation: any): boolean {
+    return (
+      billingInformation.address ||
```

Comment:
Its method should be `&&` by` ||` ?

Reply:
Is ```||``` because the idea is that at least one field has value